### PR TITLE
Add post-install warning if terra-toolkit is listed as a hard dependency

### PR DESCRIPTION
**Pull request by StephenEsser**
### Summary

This PR adds a post-install script that logs a warning message if terra-toolkit is installed as a hard dependency.

### Additional Details

It is possible for this warning to be printed multiple times if multiple versions of terra-toolkit are installed.

@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: scripts/post-install/post-install.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const { dependencies } = JSON.parse(file);
+
+  if (dependencies['terra-toolkit']) {
+    Logger.warn('+-------------------------------------------------------------------------------------------------------------+');
+    Logger.warn('|                                                   WARNING                                                   |');
+    Logger.warn('+-------------------------------------------------------------------------------------------------------------+');
+    Logger.warn('|                            terra-toolkit should be installed as a devDependency.                            |');
+    Logger.warn('|                                                                                                             |');
+    Logger.warn('| Modify the package.json to remove terra-toolkit from dependencies and add terra-toolkit to devDependencies. |');
+    Logger.warn('+-------------------------------------------------------------------------------------------------------------+');
```
Comment (StephenEsser) [inline scripts/post-install/post-install.js:21]: Formatting and warning message suggestions are welcome. The table can format strangely on small viewports, but it does help it stand out a little more in the terminal output.

File: scripts/post-install/post-install.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    throw Logger.error(error);
+  }
+
+  const { dependencies } = JSON.parse(file);
+
+  if (dependencies['terra-toolkit']) {
+    Logger.warn('+-------------------------------------------------------------------------------------------------------------+');
+    Logger.warn('|                                                   WARNING                                                   |');
+    Logger.warn('+-------------------------------------------------------------------------------------------------------------+');
+    Logger.warn('|                            terra-toolkit should be installed as a devDependency.                            |');
```
Comment (tbiethman) [inline scripts/post-install/post-install.js]: `should` -> `must`

Comment (StephenEsser) [reply scripts/post-install/post-install.js]: https://github.com/cerner/terra-toolkit/pull/326/commits/bfb2643f266a8596dbfeeccdbf18d7c05969087b

File: scripts/post-install/post-install.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const Logger = require('../utils/logger');
+
+fs.readFile(`${process.env.INIT_CWD}/package.json`, (error, file) => {
```
Comment (mjhenkes) [inline scripts/post-install/post-install.js:7]: what is the difference between `process.env.INIT_CWD` and `process.cwd()`?

Comment (StephenEsser) [reply scripts/post-install/post-install.js:7]: This is a tricky thing with npm. `process.cwd()` doesn't return the initial process directory, it returns the package being installed as the current working directory. `INIT_CWD` was added as a work around for this issue. I found a short discussion about it here: 

https://github.com/npm/npm/issues/16990

Comment (bjankord) [reply scripts/post-install/post-install.js:7]: Might be good to leave a code comment about this and link to the npm issue.

**Review by bjankord — APPROVED** [review]

File: scripts/post-install/post-install.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  if (error) {
+    throw Logger.error(error);
```
Comment (emilyrohrbough) [inline scripts/post-install/post-install.js:9]: Should we silently return null if theres an error?

Comment (StephenEsser) [reply scripts/post-install/post-install.js:9]: We can, but we might start having issues we never know about

**Comment by StephenEsser** [comment]
Here's a sample build output for anyone who is curious how the logging looks: https://travis-ci.com/cerner/terra-clinical/jobs/230813450

File: scripts/post-install/post-install.js (hunk 4 above)
Comment (emilyrohrbough) [reply scripts/post-install/post-install.js:9]: won't it come from reading the file though? I'd hate to error on people's installs

**Review by emilyrohrbough — APPROVED** [review]
I feel like this will be ignored by people in the console but changes look good

File: scripts/post-install/post-install.js (hunk 4 above)
Comment (StephenEsser) [reply scripts/post-install/post-install.js:9]: We can grab a few more opinions on this before merging. I don't mind removing the throw, but the trade off is that we will never be alerted this isn't working correctly.

**Comment by StephenEsser** [comment]
> I feel like this will be ignored by people in the console but changes look good

Hopefully at least a few people take notice of the message. We can discuss returning a non-zero exit code to fail the install on the next major version bump of terra-toolkit.

File: scripts/post-install/post-install.js (hunk 3 above)
Comment (StephenEsser) [reply scripts/post-install/post-install.js:7]: Added a short comment and reference: https://github.com/cerner/terra-toolkit/pull/326/commits/6b9636e5d7d61cd6afeab5b9fc973e7b12b1fb03

**Review by tbiethman — APPROVED** [review]

File: scripts/post-install/post-install.js (hunk 1 above)
Comment (mjhenkes) [reply scripts/post-install/post-install.js:21]: Should we state that this will fail install in the future? Even if it's just a threat? 😈

**Review by mjhenkes — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Review by nramamurth — APPROVED** [review]

**Pull request merged**